reflekt-plugin/src/main/kotlin/io/reflekt/plugin/generation/bytecode/ReflektGeneratorExtension.kt

```diff
@@ -0,0 +1,92 @@
+package io.reflekt.plugin.generation.bytecode
+
+import io.reflekt.Reflekt
+import io.reflekt.plugin.analysis.ReflektUses
+import io.reflekt.plugin.analysis.SubTypesToAnnotations
+import io.reflekt.plugin.analysis.common.ReflektName
+import io.reflekt.plugin.analysis.common.ReflektNestedName
+import io.reflekt.plugin.analysis.common.ReflektTerminalFunctionName
+import io.reflekt.plugin.analysis.common.findReflektInvokeArgumentsByExpressionPart
+import io.reflekt.plugin.analysis.psi.getFqName
+import io.reflekt.plugin.utils.Util.getUses
+import org.jetbrains.kotlin.cli.common.messages.MessageCollector
+import org.jetbrains.kotlin.codegen.StackValue
+import org.jetbrains.kotlin.codegen.asmType
+import org.jetbrains.kotlin.codegen.extensions.ExpressionCodegenExtension
+import org.jetbrains.kotlin.codegen.state.KotlinTypeMapper
+import org.jetbrains.kotlin.psi.synthetics.findClassDescriptor
+import org.jetbrains.kotlin.resolve.calls.model.ResolvedCall
+import org.jetbrains.org.objectweb.asm.Type
+import org.jetbrains.org.objectweb.asm.commons.InstructionAdapter
+
+class ReflektGeneratorExtension(private val messageCollector: MessageCollector? = null) : ExpressionCodegenExtension {
+    override fun applyFunction(receiver: StackValue, resolvedCall: ResolvedCall<*>, c: ExpressionCodegenExtension.Context): StackValue? {
```

Comment:
Please, use `messageCollector` to log some intermediate information to make debugging easier, for example before `return null` and etc